lib/chef/provisioning/convergence_strategy/ignore_convergence_failure.rb

```diff
@@ -0,0 +1,53 @@
+class Chef
+module Provisioning
+  class ConvergenceStrategy
+
+    # The purpose of this class is to decore the `converge` method with logic to catch any
+    # convergence failure exceptions, log them and then squelch them.  The reason we
+    # need this is to prevent 1 provisioned node's converge failing an entire provisioning
+    # recipe.
+    module IgnoreConvergenceFailure
+
+      attr_accessor :ignore_failures_array, :ignore_exit_values
+
+      # This module is only meant to be extended into instances, not classes or modules
```

Comment:
I don't want to extend it on the `class` level for the `install_sh` convergence strategy because users may have 1 machine with `ignore_failures` and 1 machine without it.  Thats why I extend it on the instance of the convergence_strategy - those are created per-machine.  I'll update the comment.